### PR TITLE
[JW8-10884] Typescript 4 files

### PR DESCRIPTION
**Pull request by zetagame**
### This PR will...
Type the following files:
- environment.js
- events-middleware.js
- browser-version.js
- os-version.js
### Are there any points in the code the reviewer needs to double check?
testPassive listener changed from null to noop, since addEventListener args are type-checked.

#### Addresses Issue(s):
JW8-10884

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

**Comment by zetagame** [comment]
test this please

**Comment by zetagame** [comment]
test this please

File: src/js/environment/environment.ts [hunk 1]
```diff
@@ -20,18 +20,19 @@ import {
...
     let passiveOptionRead = false;
 
     try {
-        const opts = Object.defineProperty({}, 'passive', {
+        const opts: GenericObject = Object.defineProperty({}, 'passive', {
             get: () => (passiveOptionRead = true)
         });
-        window.addEventListener('testPassive', null, opts);
-        window.removeEventListener('testPassive', null, opts);
+        window.addEventListener('testPassive', {} as EventListenerObject, opts);
```
Comment (jnatalzia) [inline src/js/environment/environment.ts]: hmmm, does this not let `null` in based on the TS "standard types"?

File: src/js/controller/events-middleware.ts [hunk 2]
```diff
@@ -1,13 +1,15 @@
-export default function middleware(model, type, currentState) {
-    let newState = currentState;
+import { GenericObject } from 'types/generic.type';
+
+export default function middleware(model: GenericObject, type: string, currentState: GenericObject): GenericObject {
```
Comment (mamaddox) [inline src/js/controller/events-middleware.ts]: Should `model` be the `CoreModel` type here?

Comment (jnatalzia) [reply src/js/controller/events-middleware.ts]: Awesome catch 👍

File: src/js/environment/environment.ts (hunk 1 above)
Comment (zetagame) [reply src/js/environment/environment.ts]: Correct!

Comment (jnatalzia) [reply src/js/environment/environment.ts]: Looks like this should be a no-op function. i get the following when passing an `{}` as the func arg in the browser
```
document.getElementById('testtest').addEventListener('click', {}, {});
TypeError: Property 'handleEvent' is not callable.
TypeError: Property 'handleEvent' is not callable.
```

Comment (robwalch) [reply src/js/environment/environment.ts]: https://github.com/Modernizr/Modernizr/issues/1894

File: src/js/environment/os-version.ts [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 }
 
-export function osVersion(osEnvironment, agent) {
-    let version;
-    let major;
-    let minor;
+export function osVersion(osEnvironment: GenericObject, agent: string): GenericObject {
+    let version: string | undefined;
+    let major: number | undefined;
+    let minor: number | undefined;
```
Comment (robwalch) [inline src/js/environment/os-version.ts]: `let major!: number;`

File: src/js/environment/environment.ts (hunk 1 above)
Comment (zetagame) [reply src/js/environment/environment.ts]: ts noop added.

File: src/js/environment/environment.ts [hunk 4]
```diff
@@ -20,18 +20,20 @@ import {
...
 
     try {
-        const opts = Object.defineProperty({}, 'passive', {
+        const opts: GenericObject = Object.defineProperty({}, 'passive', {
             get: () => (passiveOptionRead = true)
         });
-        window.addEventListener('testPassive', null, opts);
-        window.removeEventListener('testPassive', null, opts);
+        window.addEventListener('testPassive', noop, opts);
+        window.removeEventListener('testPassive', noop, opts);
```
Comment (robwalch) [inline src/js/environment/environment.ts]: We use `null` for performance https://github.com/Modernizr/Modernizr/issues/1894#issuecomment-200691178

Don't change this and use comments to ignore.

Comment (robwalch) [inline src/js/environment/environment.ts]: Disregard my earlier comment here. Modernizr settled on using `noop` for this test: 
https://github.com/Modernizr/Modernizr/blob/3675d7f6837e01a4237c91f325facf3037c030c5/feature-detects/dom/passiveeventlisteners.js

File: src/js/controller/events-middleware.ts [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
+    let newState: GenericObject = currentState;
 
     switch (type) {
         case 'time':
         case 'beforePlay':
         case 'pause':
         case 'play':
         case 'ready': {
-            const viewable = model.get('viewable');
+            const viewable: number | undefined = model.get('viewable');
```
[the remark this answers is not in the document]
Comment (zetagame) [reply src/js/controller/events-middleware.ts]: No, because it can be undefined according to annotations

**Pull request merged**